app/src/main/java/com/github/polybooks/activities/SaleInformationActivity.kt

```diff
@@ -23,6 +24,83 @@ class SaleInformationActivity: AppCompatActivity() {
         const val EXTRA_SALE_INFORMATION: String = "EXTRA_SALE_INFORMATION"
     }
 
+    private val fireStore = FirebaseFirestore.getInstance()
+    private val firebaseAuth = FirebaseAuth.getInstance()
+    private val olBookDB = OLBookDatabase { string -> url2json(string) }
```

Comment:
You should not instantiate or chose the implementation of the databases. You should use `Database.bookDatabase` instead.